packages/@react-native-windows/cli/src/runWindows/runWindows.ts

```diff
@@ -15,16 +23,54 @@ import {Command, Config} from '@react-native-community/cli-types';
 import {runWindowsOptions, RunWindowsOptions} from './runWindowsOptions';
 
 import {autoLinkCommand} from './utils/autolink';
+import {totalmem, cpus} from 'os';
+import {execSync} from 'child_process';
+
+function getDiskFreeSpace(path: string | null) : Number {
+  const out = execSync(`dir /-C ${path}`).toString().split('\r\n');
+  const line = out[out.length - 2];
+  const result = line.match(/(\d+) [^\d]+(\d+) /);
+  if (result && result.length > 2) {
+    return Number(result[2]);
+  }
+  return -1;
+}
 
-function ExitProcessWithError(loggingWasEnabled: boolean): never {
+function SetExitProcessWithError(loggingWasEnabled: boolean): void {
   if (!loggingWasEnabled) {
     console.log(
       `Re-run the command with ${chalk.bold('--logging')} for more information`,
     );
   }
-  process.exit(1);
+  process.exitCode = 1;
 }
 
+function isMSFTInternal(): boolean {
+  return process.env.USERDNSDOMAIN !== undefined && process.env.USERDNSDOMAIN.endsWith('microsoft.com');
+}
+
+function getPkgVersion(pkgName: string) : string {
+  try {
+    const pkgJson = require(`${pkgName}/package.json`);
```

Comment:
This will try to resolve from the directory of the CLI, which may not be able to resolve some of these. E.g. the CLI doesn't directly declare a dependency on RNW, so when using package managers that isolate dependencies like PNPM/Yarn 2, we won't be able to resolve it.

This is why we need special logic like this in an above file to correctly resolve the RNW package from the CLI:
```ts
    const rnwPackage = path.dirname(
      require.resolve('react-native-windows/package.json', {
        paths: [projectDir],
      }),
    );
```

As a general rule, you need to resolve starting from something that declares a dependency on what you're trying to resolve. In the above code, we start at the user project directory.

Reply:
thanks for the heads up, should be fixed now